src/bootstrap/setup.rs

```diff
@@ -86,3 +88,41 @@ d) Install Rust from source"
     };
     Ok(template.to_owned())
 }
+
+// install a git hook to automatically run tidy --bless, if they want
+fn install_git_hook_maybe(src_path: &Path) -> io::Result<()> {
+    let mut input = String::new();
+    println!(
+        "Rust's CI will automatically fail if it doesn't pass `tidy`, the internal tool for ensuring code quality.
+If you'd like, x.py can install a git hook for you that will automatically run `tidy --bless` on each commit
+to ensure your code is up to par. If you decide later that this behavior is undesirable,
+simply delete the `pre-commit` file from .git/hooks."
+    );
+
+    let should_install = loop {
+        print!("Would you like to install the git hook?: [y/N] ");
+        io::stdout().flush()?;
+        io::stdin().read_line(&mut input)?;
+        break match input.trim().to_lowercase().as_str() {
+            "y" | "yes" => true,
+            // is this the right way to check for "entered nothing"?
+            "n" | "no" | "" => false,
```

Comment:
Yes, this is right. [The docs](https://doc.rust-lang.org/std/io/trait.BufRead.html#method.read_line) say to check the number of bytes read but that doesn't work here because the input will include a newline.